docs/en/latest/plugins/ai-rag.md

```diff
@@ -0,0 +1,202 @@
+---
+title: ai-rag
+keywords:
+  - Apache APISIX
+  - API Gateway
+  - Plugin
+  - ai-rag
+description: This document contains information about the Apache APISIX ai-rag Plugin.
+---
+
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## Description
+
+The `ai-rag` plugin integrates Retrieval-Augmented Generation (RAG) capabilities with AI models.
+It allows efficient retrieval of relevant documents or information from external data sources and
+augments the AI's responses with that data, improving the accuracy and context of generated outputs.
+
+**_This plugin must be used in routes that proxy requests to LLMs only._**
+
+**_As of now only Azure OpenAI and Azure AI Search services are supported for generating embeddings and performing vector search respectively, PRs for introducing support for other service providers are welcome._**
+
+## Plugin Attributes
+
+| **Field**                                       | **Required** | **Type** | **Description**                              |
+| ----------------------------------------------- | ------------ | -------- | -------------------------------------------- |
+| embeddings_provider                             | Yes          | Object   | Configuration for the embeddings provider    |
+| embeddings_provider.azure_openai                | Yes          | Object   | Configuration for Azure OpenAI embeddings    |
+| embeddings_provider.azure_openai.endpoint       | Yes          | String   | Azure OpenAI endpoint                        |
+| embeddings_provider.azure_openai.api_key        | Yes          | String   | Azure OpenAI API key                         |
+| vector_search_provider                          | Yes          | Object   | Configuration for the vector search provider |
+| vector_search_provider.azure_ai_search          | Yes          | Object   | Configuration for Azure AI Search            |
+| vector_search_provider.azure_ai_search.endpoint | No           | String   | Azure AI Search endpoint                     |
+| vector_search_provider.azure_ai_search.api_key  | No           | String   | Azure AI Search API key                      |
```

Comment:
<img width="915" alt="image" src="https://github.com/user-attachments/assets/8efca6d2-b710-49e6-83ba-95146897b8d7">

How come the Azure AI Search config details (endpoint and api key) are optional when their parent `azure_ai_search` is required?

Reply:
oops, my bad.